src/Renode/Connectors/GPIOConnector.cs

```diff
@@ -62,15 +62,7 @@ public void SelectSourcePin(INumberedGPIOOutput source, int pinNumber)
             {
                 throw new RecoverableException("Peripheral {0} has no GPIO with number: {1}".FormatWith(source, pinNumber));
             }
-            if(sourcePin != null)
```

Comment:
While I understand that you want to have multiple sources for one output, this solution two important issues.

- if you overwrite the `sourcePin`, you will never be able to disconnect fully. You'd need to keep the collection of source pins for that.
- `OnGPIO` will set the output to the last observed value, which is not necessarily what you'd expect. I assume that for the sequence of calls "source1 - true, source2 - true, source2 - false" you'd expect the output to be `true` (as there is still one input that was set to true and not cleared). Current implementation will only remember the last input.

The correct solution would require additional changes, but there is also a simpler approach - you can easily have multiple GPIOConnectors targeting the same output. This will give you working results without changes to the code.

Reply:
So perhaps there are two issues to solve here.

1. The fact that `GPIOConnector` calls `Disconnect()` rather than passing in itself as the receiver.
2. The warning on line [70](https://github.com/renode/renode/blob/master/src/Renode/Connectors/GPIOConnector.cs#L70) which will always fire if the source pin has any connections.  This should be replaced with some check that the GPIO has a connection to the connector.